internal/server/add_document.go

```diff
@@ -0,0 +1,96 @@
+package server
+
+import (
+	"fmt"
+	"github.com/ministryofjustice/opg-sirius-supervision-deputy-hub/internal/model"
+	"github.com/ministryofjustice/opg-sirius-supervision-deputy-hub/internal/sirius"
+	"github.com/ministryofjustice/opg-sirius-supervision-deputy-hub/internal/util"
+	"mime/multipart"
+	"net/http"
+	"time"
+)
+
+type AddDocumentClient interface {
+	AddDocument(ctx sirius.Context, file multipart.File, filename string, documentType string, direction string, date string, notes string, deputyId int) error
+	GetRefData(ctx sirius.Context, refDataUrlType string) ([]model.RefData, error)
+}
+
+type AddDocumentVars struct {
+	SuccessMessage string
+	AppVars
+	DocumentDirectionRefData []model.RefData
+	DocumentTypes            []model.RefData
+	DocumentType             string
+	Direction                string
+	Date                     string
+	Notes                    string
+}
+
+func renderTemplateForAddDocument(client AddDocumentClient, tmpl Template) Handler {
+	return func(app AppVars, w http.ResponseWriter, r *http.Request) error {
+		app.PageName = "Add a document"
+
+		vars := AddDocumentVars{
+			AppVars: app,
+			Date:    time.Now().Format("2006-01-02"),
+		}
+
+		documentDirectionRefData, err := client.GetRefData(getContext(r), "/documentDirection")
+		if err != nil {
+			return err
+		}
+		vars.DocumentDirectionRefData = documentDirectionRefData
+
+		documentTypes, err := client.GetRefData(getContext(r), "?filter=noteType:deputy")
```

Comment:
If you are going to use ref data for this, you should at least make it async with goroutines. Otherwise to just load the page, the client has to fetch one set of refdata, wait for that to return, then send off a second request for the next lot.